packages/unity-interface/BrowserInterface.ts

```diff
@@ -600,7 +600,7 @@ export class BrowserInterface {
     if (message.action === FriendshipAction.REQUESTED_TO) {
       const avatar = await ensureFriendProfile(userId)
 
-      if (isAddress(userId)) {
+      if (isAddress(userId) && avatar) {
```

Comment:
This is contractually wrong, the `ensure` function should always return a valid avatar

Reply:
It seems this function can fail and is now catched, returning a lack of value. Will remove the catch from the function and let another scope handle it.